source/agora/common/BlockStorage.d

```diff
@@ -377,13 +399,76 @@ public class BlockStorage
 
     ***************************************************************************/
 
-    private void write (size_t pos, const ubyte[] data) @trusted
+    private void write (size_t pos, const ubyte[] data) nothrow @trusted
     {
-        assert(this.file !is null);
-        assert(this.getFileLength >= pos+data.length);
+        try
+        {
+            if (
+                (this.file !is null) &&
+                (pos / MapSize == this.file_index) &&
+                ((pos+data.length) / MapSize == this.file_index))
+            {
+                const size_t x0 = pos - this.file_base;
+                foreach (idx, e; data)
+                    this.file[x0+idx] = e;
+            }
+            else
+            {
+                foreach (idx, e; data)
+                    this.writeByte(pos + idx, e);
+            }
+        }
+        catch (Exception ex)
```

Comment:
Same here, why are we ignoring Exception ?

Reply:
fixed

Reply:
fixed